clients/go/events/errors/errors.go

```diff
@@ -31,6 +32,18 @@ func (r EventError) Error() string {
 	return fmt.Sprintf("%s: %s, caused by [%s]", r.Code, r.Message, r.Cause.Error())
 }
 
+func (r *EventError) Is(target error) bool {
+	t, ok := target.(*EventError)
+	if !ok {
+		return false
+	}
+	return r.Code == t.Code && (r.Cause == t.Cause || t.Cause == nil) && (r.Message == t.Message || t.Message == "")
```

Comment:
Should that just check for the Code? I feel the message and cause may be different but the errors will still be _the same_... it's not an "Equals" check.. or at least in my opinion it doesn't have to be...

Reply:
Good point. I saw it implemented this way elsewhere and thought it might add more extensibility. The idea is that when checking for an error if you don't provide the Cause or Message than it performs the check as you suggested - equality over Code and then Cause == nil / Message == "". This scenario is applied in the updates to IsNotFound, IsAlreadyExist, etc below. For example:

`errors.Is(err, &EventError{Code: ExecutionNotFound}`

Only checks if the Code is equal to any error in the stack. However, if a user wanted to include more fine-grained checks they could include a Cause or Message as well. I don't have strong feelings either way, if you prefer to perform only a Code check I can make the update.

Reply:
SG 👍🏽